libs/barista-components/table/src/table.ts

```diff
@@ -212,8 +217,11 @@ export class DtTable<T> extends _DtTableBase<T> implements OnDestroy {
   @ViewChild(CdkPortalOutlet, { static: true })
   _portalOutlet: CdkPortalOutlet;
 
-  /** @internal A list of columns */
-  @ContentChildren(DtSimpleColumnBase) _childColumns: QueryList<
+  /** @internal A list of complex columns */
+  @ContentChildren(DtHeaderCell, { descendants: true })
+  _childComplexColumns: QueryList<DtHeaderCell>;
+  /** @internal A list of complex columns */
```

Comment:
Comment is duplicated for the inverse query, please adjust.

Reply:
Will fix.